### PR TITLE
Add shape function for `aten::cross_entropy_loss`

### PR DESCRIPTION
**Pull request by vivekkhandelwal1**
Signed-Off By: Vivek Khandelwal <vivek@nod-labs.com>

### CODE REVIEW

**Review by davidberard98 — COMMENTED** [review]
can you also add a test for this one?

alternatively, I _think_ you can add this as an opinfo: 
in test_ops_jit.py, tests will run if `op.assert_jit_shape_analysis` is true. You can control that in `torch/testing/_internal/common_methods_invocations.py` (just search "assert_jit_shape_analysis" to find examples and copy it into the appropriate op, which I think is probably `nn.functional.cross_entropy`). However, in this case the op is skipped entirely for the jit test. So it might be easier to just add a separate test.

**Comment by vivekkhandelwal1** [comment]
Hi @davidberard98, could you please review this PR?

**Comment by vivekkhandelwal1** [comment]
> Hi @davidberard98, could you please review this PR?

Hi @davidberard98, I need to draw your attention to this PR. It has been pending for a long. Could you please help me get this merged?

File: test/jit/test_symbolic_shape_analysis.py [hunk 1]
```diff
@@ -573,3 +573,14 @@ def too_many_inputs(x: List[int], y: List[int], z: Any, z2: Any):
...
+    def test_cross_entropy_loss(self):
+        input = torch.randn((8, 2))
+        target = torch.randint(high=2, size=(8,))
+
+        def foo(input, target):
+            return torch.ops.aten.cross_entropy_loss(input, target)
+
+        t = torch.jit.script(foo, [input, target])
+        torch._C._jit_pass_propagate_shapes_on_graph(t.graph)
+        self.assertEqual(next(t.graph.outputs()).type().symbolic_sizes(), [])
```
Comment (davidberard98) [inline test/jit/test_symbolic_shape_analysis.py:586]: same comment as in #98144 - can we test the actual sizes?

also based on discussions with @eellison - in the future do you think would be possible to leave these new shape functions in the torch-mlir repo? we're not aware of other users who use them - and that way you don't have to wait for our reviews, and you can choose your own strategy for testing testing the registered shape functions. cc @silvasean

**Review by davidberard98 — APPROVED** [review]

File: test/jit/test_symbolic_shape_analysis.py (hunk 1 above)
Comment (silvasean) [reply test/jit/test_symbolic_shape_analysis.py]: Would you say that the symbolic shape analysis is deprecated then @eellison @davidberard98 ?

Comment (eellison) [reply test/jit/test_symbolic_shape_analysis.py]: Hmm, I don't know about deprecated, but maybe stable is a better term ? We really appreciate the previous contributions. At this point we are focusing our efforts on PT 2.0. 

I will follow up on ways torch-mlir can replicate some of the testing that we have in core for shape functions. With OpInfos and TorchDispatchMode we should be able to get pretty thorough coverage easily.

**Pull request closed without merging**